### PR TITLE
Increase kMaxEvents and error out when it's exceeded

### PR DESCRIPTION
**Pull request by paulinesho**
The current behavior silently resets and overwrites the internal list, this PR changes it to error out instead.

BUG=b/264555999

### CODE REVIEW

File: tensorflow/lite/micro/micro_profiler.cc [hunk 1]
```diff
@@ -26,7 +26,11 @@ namespace tflite {
 
 uint32_t MicroProfiler::BeginEvent(const char* tag) {
   if (num_events_ == kMaxEvents) {
-    num_events_ = 0;
+    MicroPrintf(
+        "MicroProfiler errored out because total number of events exceeded the "
+        "maximum of %d.",
+        kMaxEvents);
+    TFLITE_DCHECK(false);
```
Comment (advaitjain) [inline tensorflow/lite/micro/micro_profiler.cc:33]: nit: use `TFLITE_ASSERT_FALSE` instead?

**Review by advaitjain — COMMENTED** [review]
small nit comment.

File: tensorflow/lite/micro/micro_profiler.cc (hunk 1 above)
Comment (paulinesho) [reply tensorflow/lite/micro/micro_profiler.cc]: Yep.

**Review by advaitjain — APPROVED** [review]

**Comment by advaitjain** [comment]
@mergifyio refresh

**Review by advaitjain — APPROVED** [review]

**Pull request merged**